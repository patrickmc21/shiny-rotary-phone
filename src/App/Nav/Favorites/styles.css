.favorites-button {
  float: right;
  height: 80px;
  width: 180px;
  font-size: 28px;
}

.favorites-counter {
  margin-left: 10%;
}