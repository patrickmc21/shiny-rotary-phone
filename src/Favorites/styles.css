.favorites-button {
  border: 1px solid #66AFCF;
  border-radius: 30px;
  height: 70px;
  width: 16vw;
  background-color: transparent;
  box-shadow: 0px 0px 5px 1px #66AFCF;
  background-image: url('./images/favs.png');
  background-repeat: no-repeat;
  background-size: 90% 100%;
  background-position: -0.5vw 2px;
}

.favorites-button.active {
  border: 1px solid #F3BF44;
  box-shadow: 0px 0px 5px 1px #F3BF44;
  background-image: url('./images/favs-active.png');
}

.favorites-button:hover {
  border: 1px solid #9F1F16;
  box-shadow: 0px 0px 5px 1px #9F1F16;
  background-image: url('./images/favs-hover.png');
}

.favorites-counter {
  float: right;
  padding-top: 10px;
  height: 30px;
  width: 38px;
  color: #ffffff;
  font-size: 24px;
  font-weight: 700;
  background-image: url('./images/card-fav.png');
  background-repeat: no-repeat;
  background-size: 100% 100%;
  vertical-align: baseline;
}

