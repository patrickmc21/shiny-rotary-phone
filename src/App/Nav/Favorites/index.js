import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Favorites = () => {
  return(
    <div>I'm a Favorites button</div>
  )
}

export default Favorites;