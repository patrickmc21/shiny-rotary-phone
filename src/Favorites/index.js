import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Favorites = ({numberOfFavorites}) => {
  return(
    <button
      className='favorites-button'>
      Favorites
      <span className='favorites-counter'>{numberOfFavorites}</span>
    </button>
  )
}

Favorites.propTypes = {
  numberOfFavorites: PropTypes.number
}

export default Favorites;