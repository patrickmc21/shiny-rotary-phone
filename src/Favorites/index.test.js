import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './index.js';

describe('Favorites', () => {

  let wrapper;
  let mockNumberOfFavorites;
  let mockActivateCategory;
  let mockActiveCategoryName;

  beforeEach(() => {
    mockNumberOfFavorites = 3;
    mockActiveCategoryName = 'people';
    mockActivateCategory = jest.fn();
    wrapper = shallow(
      <Favorites 
        numberOfFavorites={mockNumberOfFavorites}
        activateCategory={mockActivateCategory}
        activeCategoryName={mockActiveCategoryName}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();  
  });

  it('should invoke activateCategory prop on click', () => {
    const expected = 'favorites'; 
    wrapper.find('.favorites-button').simulate('click');
    expect(mockActivateCategory).toHaveBeenCalledWith(expected);
  });

  it('should have an active class if favorites is selected', () => {
    mockActiveCategoryName = 'favorites';
    wrapper = shallow(<Favorites 
      numberOfFavorites={mockNumberOfFavorites}
      activateCategory={mockActivateCategory}
      activeCategoryName={mockActiveCategoryName}/>);
    expect(wrapper).toMatchSnapshot();
  });
});