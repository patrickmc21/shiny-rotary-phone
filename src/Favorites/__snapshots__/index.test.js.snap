// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should have an active class if favorites is selected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    activateCategory={[Function]}
    activeCategoryName="favorites"
    numberOfFavorites={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="favorites-counter"
>
        3
</span>,
      "className": "favorites-button active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 3,
        "className": "favorites-counter",
      },
      "ref": null,
      "rendered": 3,
      "type": "span",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="favorites-counter"
>
          3
</span>,
        "className": "favorites-button active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 3,
          "className": "favorites-counter",
        },
        "ref": null,
        "rendered": 3,
        "type": "span",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Favorites should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    activateCategory={[Function]}
    activeCategoryName="people"
    numberOfFavorites={3}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="favorites-counter"
>
        3
</span>,
      "className": "favorites-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 3,
        "className": "favorites-counter",
      },
      "ref": null,
      "rendered": 3,
      "type": "span",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="favorites-counter"
>
          3
</span>,
        "className": "favorites-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 3,
          "className": "favorites-counter",
        },
        "ref": null,
        "rendered": 3,
        "type": "span",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
